### ISSUE TITLE
`nodetool removenode` may fail if we concurrently try to do `DROP INDEX...` resulting in leaving a dead DB node as part of the DB cluster

### ISSUE THREAD

**Issue by vponomaryov** [scylladb/scylladb#18436]
## Packages

Scylla version: `5.5.0~dev-20240411.0be61e51d329` with build-id `54e018bb43eb2ae707aac127097747cd39532532`

Kernel Version: `5.15.0-1057-aws`

## Issue description

- [ ] This issue is a regression.
- [ ] It is unknown if this issue is a regression.

In this scenario we were running 2 nemesis threads in parallel.
One thread was running `disrupt_remove_node_then_add_node` nemesis and another one `disrupt_create_index`.
When the first nemesis was running the `nodetool removenode` op the second one dropped index that it had created before the first nemesis.
It led to the abort of the `removenode` operation agreed by alive nodes:
```
2024-04-13 05:01:31,822 f:common.py       l:1823 c:utils                p:DEBUG > Executing CQL 'CREATE INDEX test_v_nemesis ON scylla_bench.test("v")' ...
...
2024-04-13 05:16:54,847 f:remote_base.py  l:521  c:RemoteLibSSH2CmdRunner p:DEBUG > Running command "/usr/bin/nodetool  removenode e35af705-73fa-4dd1-8fbc-7ef886511ade "...
...
2024-04-13 05:20:37,095 f:common.py       l:1831 c:utils                p:DEBUG > Executing CQL '<SimpleStatement query="DROP INDEX scylla_bench.test_v_nemesis", consistency=Not Set>' ...
...
2024-04-13 05:20:53,338 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > Datacenter: eu-west
2024-04-13 05:20:53,338 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > ===================
2024-04-13 05:20:53,338 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > Status=Up/Down
2024-04-13 05:20:53,338 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > |/ State=Normal/Leaving/Joining/Moving
2024-04-13 05:20:53,338 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > -- Address     Load     Tokens Owns Host ID                              Rack
2024-04-13 05:20:53,338 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > UN 10.4.11.237 31.32 GB 256    ?    24ca87a5-b1af-4a30-81ed-65d33571caf8 1c  
2024-04-13 05:20:53,338 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > UN 10.4.8.169  27.93 GB 256    ?    b3695113-7844-4d5d-86ff-d001329ea0d8 1c  
2024-04-13 05:20:53,338 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > UN 10.4.8.224  28.41 GB 256    ?    45760ec2-0a80-4ec9-8b24-663b413ba961 1c  
2024-04-13 05:20:53,338 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > UN 10.4.8.26   34.22 GB 256    ?    3185ce37-9ee7-4de5-8fde-3ec1ca5c701d 1c  
2024-04-13 05:20:53,338 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > DN 10.4.9.250  21.90 GB 256    ?    e35af705-73fa-4dd1-8fbc-7ef886511ade 1c
``` 

Nodes logs:
`<node-3>`:
```
Apr 13 05:20:51.964572 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 0:strm] storage_service - aborting node operation ops_uuid=e603c4dc-685c-435e-a343-f125e4581815
Apr 13 05:20:51.964731 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 0:strm] storage_service - removenode[e603c4dc-685c-435e-a343-f125e4581815]: Removed node=10.4.9.250 as leaving node, coordinator=10.4.8.26
Apr 13 05:20:51.990290 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 0:strm] node_ops - removenode[e603c4dc-685c-435e-a343-f125e4581815]: The removenode_abort command failed while handling a previous error, sync_nodes={10.4.8.224, 10.4.11.237, 10.4.8.169, 10.4.8.26}: std::runtime_error (The removenode_abort command failed for nodes={10.4.8.169}). Ignoring
```

`<node-4>`:
```
Apr 13 05:20:51.965141 longevity-parallel-topology-schema--db-node-4bdf0ea1-4 scylla[7336]:  [shard 0:strm] storage_service - aborting node operation ops_uuid=e603c4dc-685c-435e-a343-f125e4581815
Apr 13 05:20:51.965291 longevity-parallel-topology-schema--db-node-4bdf0ea1-4 scylla[7336]:  [shard 0:strm] storage_service - removenode[e603c4dc-685c-435e-a343-f125e4581815]: Removed node=10.4.9.250 as leaving node, coordinator=10.4.8.26
```

`<node-5>`:
```
2024-04-13T05:20:52.281+00:00 longevity-parallel-topology-schema--db-node-4bdf0ea1-5     !INFO | scylla[7374]:  [shard 0:strm] storage_service - aborting node operation ops_uuid=e603c4dc-685c-435e-a343-f125e4581815
2024-04-13T05:20:52.281+00:00 longevity-parallel-topology-schema--db-node-4bdf0ea1-5     !INFO | scylla[7374]:  [shard 0:strm] storage_service - removenode[e603c4dc-685c-435e-a343-f125e4581815]: Removed node=10.4.9.250 as leaving node, coordinator=10.4.8.26
```

`<node-7>`:
```
Apr 13 05:20:51.964691 longevity-parallel-topology-schema--db-node-4bdf0ea1-7 scylla[7304]:  [shard 0:strm] storage_service - aborting node operation ops_uuid=e603c4dc-685c-435e-a343-f125e4581815
Apr 13 05:20:51.964815 longevity-parallel-topology-schema--db-node-4bdf0ea1-7 scylla[7304]:  [shard 0:strm] storage_service - removenode[e603c4dc-685c-435e-a343-f125e4581815]: Removed node=10.4.9.250 as leaving node, coordinator=10.4.8.26
```

Then one of the following nemesis failed with the following error:
```
error executing POST request to http://localhost:10000/storage_service/decommission with parameters {}: remote replied with status code 500 Internal Server Error:
std::runtime_error (decommission[b7236c1a-106e-431e-bfe2-175e4b789ad9]: \
  Cannot start: nodes={10.4.9.250} needed for decommission operation are down. It is highly recommended to fix the down nodes and try again.)
```

The `10.4.9.250` is the dead node left by the `remove_node_then_add_node` nemesis.

## Impact

Spoils whole test run.

## How frequently does it reproduce?

First time observed

## Installation details

Cluster size: 5 nodes (i4i.2xlarge)

Scylla Nodes used in this run:
- longevity-parallel-topology-schema--db-node-4bdf0ea1-8 (3.249.163.150 | 10.4.8.8) (shards: 7)
- longevity-parallel-topology-schema--db-node-4bdf0ea1-7 (54.170.247.145 | 10.4.8.169) (shards: 7)
- longevity-parallel-topology-schema--db-node-4bdf0ea1-6 (54.216.131.244 | 10.4.9.28) (shards: -1)
- longevity-parallel-topology-schema--db-node-4bdf0ea1-5 (3.255.153.131 | 10.4.8.224) (shards: 7)
- longevity-parallel-topology-schema--db-node-4bdf0ea1-4 (34.255.216.239 | 10.4.11.237) (shards: 7)
- longevity-parallel-topology-schema--db-node-4bdf0ea1-3 (34.244.101.250 | 10.4.8.26) (shards: 7)
- longevity-parallel-topology-schema--db-node-4bdf0ea1-2 (18.201.146.10 | 10.4.9.250) (shards: 7)
- longevity-parallel-topology-schema--db-node-4bdf0ea1-1 (3.255.195.175 | 10.4.8.111) (shards: 7)

OS / Image: `ami-03a7123d7bfc23027` (aws: undefined_region)

Test: `longevity-schema-topology-changes-12h-test`
Test id: `4bdf0ea1-8890-468e-9b6e-c89d5583acef`
Test name: `scylla-master/tier1/longevity-schema-topology-changes-12h-test`
Test config file(s):

- [longevity-parallel-schema-changes-12h.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/ead2529c1f5f3816f04e3b23de737a359134c053/test-cases/longevity/longevity-parallel-schema-changes-12h.yaml)

<details>
<summary>
Logs and commands
</summary>

- Restore Monitor Stack command: `$ hydra investigate show-monitor 4bdf0ea1-8890-468e-9b6e-c89d5583acef`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=4bdf0ea1-8890-468e-9b6e-c89d5583acef)
- Show all stored logs command: `$ hydra investigate show-logs 4bdf0ea1-8890-468e-9b6e-c89d5583acef`

## Logs:
- **db-cluster-4bdf0ea1.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/4bdf0ea1-8890-468e-9b6e-c89d5583acef/20240413_062353/db-cluster-4bdf0ea1.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/4bdf0ea1-8890-468e-9b6e-c89d5583acef/20240413_062353/db-cluster-4bdf0ea1.tar.gz)
- **sct-runner-events-4bdf0ea1.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/4bdf0ea1-8890-468e-9b6e-c89d5583acef/20240413_062353/sct-runner-events-4bdf0ea1.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/4bdf0ea1-8890-468e-9b6e-c89d5583acef/20240413_062353/sct-runner-events-4bdf0ea1.tar.gz)
- **sct-4bdf0ea1.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/4bdf0ea1-8890-468e-9b6e-c89d5583acef/20240413_062353/sct-4bdf0ea1.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/4bdf0ea1-8890-468e-9b6e-c89d5583acef/20240413_062353/sct-4bdf0ea1.log.tar.gz)
- **loader-set-4bdf0ea1.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/4bdf0ea1-8890-468e-9b6e-c89d5583acef/20240413_062353/loader-set-4bdf0ea1.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/4bdf0ea1-8890-468e-9b6e-c89d5583acef/20240413_062353/loader-set-4bdf0ea1.tar.gz)
- **monitor-set-4bdf0ea1.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/4bdf0ea1-8890-468e-9b6e-c89d5583acef/20240413_062353/monitor-set-4bdf0ea1.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/4bdf0ea1-8890-468e-9b6e-c89d5583acef/20240413_062353/monitor-set-4bdf0ea1.tar.gz)

[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-master/job/tier1/job/longevity-schema-topology-changes-12h-test/9/)
[Argus](https://argus.scylladb.com/test/e17ff39a-f3f6-4ac6-a161-871267fd3955/runs?additionalRuns[]=4bdf0ea1-8890-468e-9b6e-c89d5583acef)
</details>

**Comment by mykaul** [comment]
So we have a dead node. Can you remove it? I'm not sure I fully understand the impact.

**Comment by vponomaryov** [comment]
> So we have a dead node. Can you remove it? I'm not sure I fully understand the impact.

It is unknown whether it is possible to remove it later or not.
It requires either manual intervention when bug appears or enhancement to automation to cover this case.

**Comment by mykaul** [comment]
@kbr-scylla - is this something your team should look at?

**Comment by kbr-scylla** [comment]
The fact that a failed removenode spoils the rest of the SCT run because it tried to do something else to the dead node (which was supposed to be removed) -- this is not something that we can help with in ScyllaDB implementation.

Removenode can fail to any number of reasons, for example, a transient connectivity issue leading to a timeout leading to a failure. So there are reasons that we don't have control over -- problems with network or environment in general.

The SCT automation is the only place where you can deal with such issues somehow. For example, by retrying removenode.

---

The actual issue inside the ScyllaDB product that we can investigate and perhaps find a bug which can then be fixed -- is **why** removenode failed in this particular case, was it somehow connected to the concurrently running DROP INDEX? Now that is something that we can focus on, and it should be also the focus of the report, not the fact that SCT run is spoiled.

The fact that a dead node continues to be part of the cluster after a failed removenode is perfectly normal, after all, the operation failed.

**Comment by kbr-scylla** [comment]
Failure happened here:
```
Apr 13 05:20:48.453347 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 0:strm] node_ops - removenode[e603c4dc-685c-435e-a343-f125e4581815]: Node 10.4.8.169 failed removenode_sync_data verb: seastar::rpc::remote_verb_error ({shard 0: std::runtime_error (repair[7c1d569e-55dc-44df-a701-44028be68f69]: 2 out of 576 ranges failed, keyspace=scylla_bench, tables={test_v_nemesis_index, test, test_counters}, repair_reason=removenode, nodes_down_during_repair={}, aborted_by_user=false, failed_because=seastar::abort_requested_exception (abort requested)), shard 1: std::runtime_error (repair[7c1d569e-55dc-44df-a701-44028be68f69]: 3 out of 576 ranges failed, keyspace=scylla_bench, tables={test_v_nemesis_index, test, test_counters}, repair_reason=removenode, nodes_down_during_repair={}, aborted_by_user=false, failed_because=seastar::abort_requested_exception (abort requested)), shard 2: std::runtime_error (repair[7c1d569e-55dc-44df-a701-44028be68f69]: 2 out of 576 ranges failed, keyspace=scylla_bench, tables={test_v_nemesis_index, test, test_counters}, repair_reason=removenode, nodes_down_during_repair={}, aborted_by_user=false, failed_because=seastar::abort_requested_exception (abort requested)), shard 3: std::runtime_error (repair[7c1d569e-55dc-44df-a701-44028be68f69]: 2 out of 576 ranges failed, keyspace=scylla_bench, tables={test_v_nemesis_index, test, test_counters}, repair_reason=removenode, nodes_down_during_repair={}, aborted_by_user=false, failed_because=seastar::abort_requested_exception (abort requested)), shard 5: std::runtime_error (repair[7c1d569e-55dc-44df-a701-44028be68f69]: 5 out of 576 ranges failed, keyspace=scylla_bench, tables={test_v_nemesis_index, test, test_counters}, repair_reason=removenode, nodes_down_during_repair={}, aborted_by_user=false, failed_because=seastar::abort_requested_exception (abort requested))})
```

`abort_requested_exception`?

Rings a bell: https://github.com/scylladb/scylladb/issues/18452#issuecomment-2095532313

**Comment by kbr-scylla** [comment]
Nodes indeed are trying to sync schema for some time:
```
Apr 13 05:20:44.263074 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 1:strm] migration_manager - Pulling schema from 10.4.8.169:0
Apr 13 05:20:44.270270 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 3:strm] migration_manager - Requesting schema pull from 10.4.8.169:0
Apr 13 05:20:44.270285 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 3:strm] migration_manager - Requesting schema pull from 10.4.8.169:0
Apr 13 05:20:44.270290 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 3:strm] migration_manager - Requesting schema pull from 10.4.8.169:0
Apr 13 05:20:44.272722 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 2:comp] compaction - [Compact system.truncated 93c9db00-f955-11ee-96de-0e6264c12f55] Compacted 3 sstables to [/var/lib/scylla/data/system/truncated-38c19fd0fb863310a4b70d0cc66628aa/me-3gf9_0euk_198342aldmci9cdowl-big-Data.db:level=0]. 16kB to 6052 bytes (~35% of original) in 44ms = 383kB/s. ~384 total partitions merged to 1.
Apr 13 05:20:44.274520 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 2:comp] compaction - [Compact system.truncated 93d3ed20-f955-11ee-96de-0e6264c12f55] Compacting [/var/lib/scylla/data/system/truncated-38c19fd0fb863310a4b70d0cc66628aa/me-3gf9_0euk_168282aldmci9cdowl-big-Data.db:level=0:origin=memtable,/var/lib/scylla/data/system/truncated-38c19fd0fb863310a4b70d0cc66628aa/me-3gf9_0euk_198342aldmci9cdowl-big-Data.db:level=0:origin=compaction]
Apr 13 05:20:44.281341 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 1:strm] migration_manager - Requesting schema pull from 10.4.8.169:0
Apr 13 05:20:44.281356 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 1:strm] migration_manager - Requesting schema pull from 10.4.8.169:0
Apr 13 05:20:44.282116 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 1:strm] migration_manager - Requesting schema pull from 10.4.8.169:0
Apr 13 05:20:44.304522 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 0:strm] schema_tables - Schema version changed to 8f8df4ae-f955-11ee-b786-7bde3135826c
Apr 13 05:20:44.305280 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 3:strm] migration_manager - Schema merge with 10.4.8.169:0 completed
Apr 13 05:20:44.305362 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 3:strm] migration_manager - Pulling schema from 10.4.8.169:0
Apr 13 05:20:44.309879 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 3:strm] migration_manager - Requesting schema pull from 10.4.8.169:0
Apr 13 05:20:44.321474 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 2:comp] compaction - [Compact system.truncated 93d3ed20-f955-11ee-96de-0e6264c12f55] Compacted 2 sstables to [/var/lib/scylla/data/system/truncated-38c19fd0fb863310a4b70d0cc66628aa/me-3gf9_0euk_1ndcg2aldmci9cdowl-big-Data.db:level=0]. 11kB to 5930 bytes (~51% of original) in 41ms = 282kB/s. ~256 total partitions merged to 1.
Apr 13 05:20:44.327695 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 1:strm] migration_manager - Requesting schema pull from 10.4.8.169:0
Apr 13 05:20:44.337712 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 3:strm] migration_manager - Requesting schema pull from 10.4.8.169:0
Apr 13 05:20:44.348305 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 3:strm] migration_manager - Requesting schema pull from 10.4.8.169:0
Apr 13 05:20:44.349917 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 0:strm] schema_tables - Schema version changed to 8f8df4ae-f955-11ee-b786-7bde3135826c
Apr 13 05:20:44.350780 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 1:strm] migration_manager - Schema merge with 10.4.8.169:0 completed
Apr 13 05:20:44.350805 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 1:strm] migration_manager - Pulling schema from 10.4.8.169:0
Apr 13 05:20:44.368471 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 0:comp] compaction - [Compact system.peers 93e24500-f955-11ee-9f04-0e6664c12f55] Compacting [/var/lib/scylla/data/system/peers-37f71aca7dc2383ba70672528af04d4f/me-3gf9_0euk_1x0kg2f1z02gz2009h-big-Data.db:level=0:origin=memtable,/var/lib/scylla/data/system/peers-37f71aca7dc2383ba70672528af04d4f/me-3gf9_0die_1inlc2f1z02gz2009h-big-Data.db:level=0:origin=compaction]
Apr 13 05:20:44.374145 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 1:strm] migration_manager - Requesting schema pull from 10.4.8.169:0
Apr 13 05:20:44.378543 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 1:strm] migration_manager - Requesting schema pull from 10.4.8.169:0
Apr 13 05:20:44.436832 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 0:strm] schema_tables - Schema version changed to 8f8df4ae-f955-11ee-b786-7bde3135826c
Apr 13 05:20:44.438296 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 3:strm] migration_manager - Schema merge with 10.4.8.169:0 completed
Apr 13 05:20:44.438322 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 3:strm] migration_manager - Pulling schema from 10.4.8.169:0
Apr 13 05:20:44.447989 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 0:comp] compaction - [Compact system.peers 93e24500-f955-11ee-9f04-0e6664c12f55] Compacted 2 sstables to [/var/lib/scylla/data/system/peers-37f71aca7dc2383ba70672528af04d4f/me-3gf9_0euk_27y342f1z02gz2009h-big-Data.db:level=0]. 42kB to 30kB (~72% of original) in 67ms = 636kB/s. ~256 total partitions merged to 6.
Apr 13 05:20:44.449635 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 0:comp] compaction - [Compact system.scylla_table_schema_history 93eea110-f955-11ee-9f04-0e6664c12f55] Compacting [/var/lib/scylla/data/system/scylla_table_schema_history-0191a53e40f031d49171b0d19ffb17b4/me-3gf9_0euk_26nsg2f1z02gz2009h-big-Data.db:level=0:origin=memtable,/var/lib/scylla/data/system/scylla_table_schema_history-0191a53e40f031d49171b0d19ffb17b4/me-3gf9_0die_1yav42f1z02gz2009h-big-Data.db:level=0:origin=compaction]
Apr 13 05:20:44.487224 longevity-parallel-topology-schema--db-node-4bdf0ea1-3 scylla[7325]:  [shard 3:strm] migration_manager - Requesting schema pull from 10.4.8.169:0
```

but wait... Why are there schema pulls? We're in Raft mode... This code should be disabled

**Comment by kbr-scylla** [comment]
Ok, this:
```cpp
    if (schema_map.empty()) {
        co_return;
    }
    if (!_enable_schema_pulls) {
        co_await _group0_barrier.trigger(_as);
        co_return;
    }
```

was added in c6c83474939. The run used 0be61e51d329 which was before

**Comment by kbr-scylla** [comment]
According to node-3 the abort_requested originated at node-7, and there we can see: 
```
Apr 13 05:20:47.635055 longevity-parallel-topology-schema--db-node-4bdf0ea1-7 scylla[7304]:  [shard 2:strm] repair - repair[7c1d569e-55dc-44df-a701-44028be68f69]: 2 out of 576 ranges failed, keyspace=scylla_bench, tables={test_v_nemesis_index, test, test_counters}, repair_reason=removenode, nodes_down_during_repair={}, aborted_by_user=false, failed_because=seastar::abort_requested_exception (abort requested)
```

The table being repaired is `test_v_nemesis_index` --- let me guess, that's the index being dropped

So probably dropping an index that's being repaired cancels the repair

**Comment by kbr-scylla** [comment]
Heh, isn't this just https://github.com/scylladb/scylladb/issues/15598, except we're getting `abort_requested_exception` instead of `no_such_column_family` for some reason? cc @Deexie @denesb

**Comment by kbr-scylla** [comment]
@Deexie why does the regression test in 2ea5d9b62327eb20727205d534f8112558518c45 disable repair-based ops? repair-based ops are the default

**Comment by kbr-scylla** [comment]
I suspect the exception might be flying out of this `abort_on_expiry` here:
```cpp
future<table_dropped> table_sync_and_check(replica::database& db, service::migration_manager& mm, const table_id& uuid) {
    if (mm.use_raft()) {
        abort_on_expiry aoe(lowres_clock::now() + std::chrono::seconds{10});
        auto& as = aoe.abort_source();
        auto sub = mm.get_abort_source().subscribe([&as] () noexcept {
            if (!as.abort_requested()) {
                as.request_abort();
            }
        });

        // Trigger read barrier to synchronize schema.
        co_await mm.get_group0_barrier().trigger(as);
    }

    co_return !db.column_family_exists(uuid);
}
```

**Comment by kbr-scylla** [comment]
I wonder whether this DROP INDEX actually finishes, or perhaps it's waiting for something. And if it finishes -- how long does it take. It seems plausible that we reach the 10 seconds timeout if DROP INDEX takes some time to finish and so the read barrier has to wait for it. And then there's the fact that multiple repair tasks on multiple shards could all be calling read_barrier at the same time and some of these barriers get serialized due to how `serialized_action _group0_barrier` across shards

**Comment by kbr-scylla** [comment]
Closing as duplicate of https://github.com/scylladb/scylladb/issues/18452.
